src/ImageSharp/Formats/Jpeg/Components/Decoder/ColorConverters/JpegColorConverter.FromYccKVector8.cs

```diff
@@ -18,34 +18,23 @@ public FromYccKVector8(int precision)
             {
             }
 
-            protected override void ConvertCoreVectorized(in ComponentValues values, Span<Vector4> result)
+            protected override void ConvertCoreVectorizedInplace(in ComponentValues values)
             {
-                ref Vector<float> yBase =
+                ref Vector<float> c0Base =
                     ref Unsafe.As<float, Vector<float>>(ref MemoryMarshal.GetReference(values.Component0));
-                ref Vector<float> cbBase =
+                ref Vector<float> c1Base =
                     ref Unsafe.As<float, Vector<float>>(ref MemoryMarshal.GetReference(values.Component1));
-                ref Vector<float> crBase =
+                ref Vector<float> c2Base =
                     ref Unsafe.As<float, Vector<float>>(ref MemoryMarshal.GetReference(values.Component2));
                 ref Vector<float> kBase =
                     ref Unsafe.As<float, Vector<float>>(ref MemoryMarshal.GetReference(values.Component3));
 
-                ref Vector4Octet resultBase =
-                    ref Unsafe.As<Vector4, Vector4Octet>(ref MemoryMarshal.GetReference(result));
-
                 var chromaOffset = new Vector<float>(-this.HalfValue);
 
                 // Walking 8 elements at one step:
-                int n = result.Length / 8;
-
-                Vector4Pair rr = default;
-                Vector4Pair gg = default;
-                Vector4Pair bb = default;
-
-                ref Vector<float> rrRefAsVector = ref Unsafe.As<Vector4Pair, Vector<float>>(ref rr);
-                ref Vector<float> ggRefAsVector = ref Unsafe.As<Vector4Pair, Vector<float>>(ref gg);
-                ref Vector<float> bbRefAsVector = ref Unsafe.As<Vector4Pair, Vector<float>>(ref bb);
+                int n = values.Component0.Length / 8;
 
-                var scale = new Vector<float>(1 / this.MaximumValue);
+                var scale = new Vector<float>(1 / (this.MaximumValue * this.MaximumValue));
                 var max = new Vector<float>(this.MaximumValue);
```

Comment:
```suggestion
                var max = new Vector<float>(this.MaximumValue);
                var scale = new Vector<float>(1f) / (max * max);
```
is [less ceremony](https://sharplab.io/#v2:EYLgxg9gTgpgtADwGwBYA0AXEBDAzgWwB8ABAJgEYBYAKGIAYACY8gOgCUBXAOwwEt8YLAMIR8AB14AbGFADKMgG68wMXAG4a9JqwByHAVGXqamgMxNSDIQwDeNBg6bmAZpIjYMDALLYE/fQBq2JIcMLYMAOYwGGoMuNGxAL72jikOYoYKHmEBMGAY0AA8ru4YAHwMAPq4YMEwGtSODBm8WRg5eQVQxW4eFZX4vg1NaU5MKAwAggAUAJSjdo1NjtW10gwAvAxcMADuDLn5RSV90+QMAPQM0xgAFry4LD5++IHBoQwAVAx3D0++/nwQRCMFms2GywcA18m22ewOnWOvXKN3uj2egOBoXBo2SSwco2I5mIEwAQnMFqMmlkoAxBghYTt9ocuj1SmVUX8Ma8ge9QRDIdCGVt6QLllTHDS4mswlsmQijt0TijyM4cfjlqs6rCatqrtN6V86b51U08YkgA).